### ISSUE TITLE
WMS data - checkable features and group of data

### ISSUE THREAD

**Issue by cath222ou** [qgis/QGIS#45968]
### Feature description

When using WMS data, we would like to see the data like in ArcMap where each class of the WMS service is checkable and the groups are preserved.

For example, in ArcMap, I get this with my WMS service:
![image](https://user-images.githubusercontent.com/30536406/140968013-5e163bb3-9733-45fc-ae89-752db0d48cad.png)

But, with QGIS, I obtain this:
![image](https://user-images.githubusercontent.com/30536406/140968032-434ed832-18da-4ac9-ba32-39605d813268.png)

In QGIS, we have lost the class grouping. Also, we can't decide which classes we want to check/uncheck for visualization.

**Comment by gioman** [comment]
> my WMS service

@cath222ou can you share the URL?

**Comment by cath222ou** [comment]
@gioman Unfortunately, I can't share the URL because it is a private URL. However, I can share with you another WMS URL that has the same result as my WMS service in ArcMap and QGIS.

https://geoegl.msp.gouv.qc.ca/ws/mffpecofor.fcgi

Example: 

In ArcMap:

![image](https://user-images.githubusercontent.com/30536406/140977832-317ba176-51ff-4b93-ad4e-bc54dd303ba3.png)

In QGIS:
![image](https://user-images.githubusercontent.com/30536406/140977887-b81eb25d-9b49-48d3-b885-03a3bf21ce2f.png)

**Comment by gioman** [comment]
> However, I can share with you another WMS URL

@cath222ou I agree, that would be immensely useful. I edited the title to better reflect the request.

**Comment by ismailsunni** [comment]
Probably related to this issue https://github.com/qgis/QGIS/issues/44107
